### DISCUSSION TITLE
Upgrade scripts fails in proxy environments. Add -E to sudo?

### DISCUSSION THREAD

**Discussion by torstefan** [wifinigel/wiperf#59] [Bugs - I think I may have found a bug]
Hey

Is it possible for the setup.sh scrip and the guids to be updated with sudo -E ? Makes it so that the subprocess inherit the enviroment with the proxy settings.

Kind regards,
Tor Stefan

root@snorty-svg:/var/log# curl -s https://raw.githubusercontent.com/wifinigel/wiperf/dev/setup.sh | sudo bash -s upgrade rpi
(ok) wiperf will now be unistalled, then re-installed with a new version
(ok) checking we can get to the Internet before we start...
(fail) Unable to reach Internet - check connection (exiting)

root@snorty-svg:~# curl -s https://raw.githubusercontent.com/wifinigel/wiperf/dev/setup.sh | sudo -E bash -s upgrade rpi
(ok) wiperf will now be unistalled, then re-installed with a new version
(ok) checking we can get to the Internet before we start...
(ok) Starting wiperf uninstall process (see /var/log/wiperf_install.log for details)
(ok) Removing Python modules
(ok) ...wiperf_poller
(ok) Removing install dir
(ok) Removing config dir
(ok) Removing switcher script
(ok) Removing log files
(ok) Done
(ok) Installing latest version of wiperf...
(ok) Starting wiperf install process for rpi (see /var/log/wiperf_install.log for details)
(ok) Version: 2.2.0 (poller ver: 0.3.0, branch: main)
Unable to reach Internet - check connection (exiting)

**Comment by wifinigel** [comment]
Hi,

I can add a note to the documentation to advise users that they may use this option if they need it?

I don't think there is anything to change in the wiperf code though? (unless I mis-understand).

Nigel.

**Reply by torstefan** [reply]
Would be nice to change the setup.sh code on line 307, add an -E to the sudo

So you call sudo in all the instances with an -E

I think its two places, the first in the documentation, and the second is in the setup.sh

  # install new code
  echo "(ok) Installing latest version of wiperf..."
  curl -s https://raw.githubusercontent.com/wifinigel/wiperf/${GITHUB_BRANCH}/setup.sh | sudo -E bash -s install $PLATFORM

**Reply by wifinigel** [reply]
OK, this sounds like a good suggestion. I will add this to the list of things to add to wiperf :)
